libraries/FSTools/FSTools.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <FS.h> 
+#include <memory>
+#include <functional>
+/*
+ 
+	A temporary FS is made between the END of the sketch...  and the start of the partition you try to mount, to maximise the available space for copying the FS. 
+	The WORST case this is at 0x40300000 which is for a 3m FS on 4m flash..  leaving 460Kb for copying. 
+
+
+	I have not worked out a way to prevent linking of ALL the layouts except to use #ifdef guards.  Ideas welcome as it uses 400B ROM. 
+ */
+
+namespace FST {
+
+struct layout {
+  layout(uint32_t s, uint32_t e, uint32_t p, uint32_t b) : startAddr(s), endAddr(e), page(p), block(b) {};
+  uint32_t startAddr{0};
+  uint32_t endAddr{0};
+  uint32_t page{0};
+  uint32_t block{0};
+};
+
+enum FS_t : uint8_t {
+	SPIFFS,
+	LITTLEFS
+};
+
+extern const layout layout_512k32;
```

Comment:
To save the ROM space, is it possible to make these layouts `constexpr layout layout_512k32 = {0x40273000, 0x4027B000, 0x100, 0x1000};`?

Reply:
Making these changes and also making the struct constructer `constexpr` seems to save ~500 bytes, so thank you for that.  I've not objdump'd the elf to check though.